syft/formats/formats.go

```diff
@@ -71,6 +81,13 @@ func ByNameAndVersion(name string, version string) sbom.Format {
 	for _, f := range Formats() {
 		for _, n := range f.IDs() {
 			if cleanFormatName(string(n)) == name && versionMatches(f.Version(), version) {
+				if version == sbom.AnyVersion && strings.Contains(string(n), "cyclonedx") {
```

Comment:
This is looking for the string `cyclonedx` but we might introduce an alias like `cdx`. I think this should more formally ask the format for the latest supported version. 

At the same time, I feel that this format object is about to be rethought anyway. I think there are strong signs to split up for encoding vs decoding use cases, which means trying to settle this detail in this PR wouldn't be worth it. I think it's worth at least adding a comment here about the limitations of this approach and a TODO to refactor later.

Reply:
I would probably like to add a `Default` field. At some point in the future, when we support SPDX 3, we will probably also want to continue to default to SPDX 2.3.

Reply:
I can follow this PR up with a stab at that. I really want to keep this PR small and focused on integrating the new cyclonedx library. Other changes on how we handle formats, their defaults or how the latest is selected can be their own change set. The reason this line had to be included was because we don't necessarily want to be on latest for cyclonedx at this moment. Rather than rewrite ByNameAndVersion into a totally new thing I thought a simple edge case with comment would be the correct change here